### DISCUSSION TITLE
Livewire \ Exceptions \ CorruptComponentPayloadException with Route Model Binding

### DISCUSSION THREAD

**Discussion by d-shannon** [livewire/livewire#4743] [7. Bugs (archived)] [answered]
Hi All,

Wondering if anyone else is experiencing  CorruptComponentPayloadException when using route model binding.

"Livewire encountered corrupt data when trying to hydrate the [.....] component....."

**Laravel 9.2
Livewire 2.10.4**

```
<?php

namespace App\Http\Livewire\Dashboard\Project;

use App\Models\Project;
use Livewire\Component;

class Notifications extends Component
{

    public Project $project;

    public function submit()
    {
    }

}
```

Component:

`<div>

    <x-button wire:click="submit">Test</x-button>

</div>`

Thanks!

**Comment by joshhanley** [comment]
@d-shannon I've never had an issue with it, route model binding seems to work happily for me. Just as a test, can you remove the `Project` type hint and see if the error still happens?

**Reply by d-shannon** [reply]
Hi Josh,

Thanks for the reply! If I remove the type hint and click the button, no error. If I try the following method I also get an error. Any ideas how to work around it so I can use the model?

```
   public $project;
    
    public function mount(Project $project)
    {
        $this->project =  $project;
    }
```

**Comment by d-shannon** [comment]
I found out what was causing the issue @joshhanley!

I am using snowflakes [godruoyi/php-snowflake](https://github.com/godruoyi/php-snowflake) instead of incrementing IDs. Turns out livewire doesn't like them.

If there is only one record in the table, no exception is thrown. If there is more than one record in the table an exception is thrown.

If I switch back to auto incrementing ids, everything works as expected.

**Comment by joshhanley** [comment, marked as answer]
@d-shannon ah right did't realise you were using that! I was actually helping someone with this issue with snowflakes on the Livewire discord yesterday!

So to explain why they don't work, it's because you are running up against Javascript's integer limit 

![image](https://user-images.githubusercontent.com/882837/156272482-cda36b1d-595c-4590-a8f1-db2ca1c72b7b.png)

The only way to get around that is to ensure that all snowflake IDs are cast to a string before adding them to any of your components public properties. That way there is no corruption of the data 🙂

**Reply by d-shannon** [reply]
Genius! Thanks for an explanation. This was driving me crazy. Now working with your solution! Cheers


```
    public function booted()
    {
        $this->project->id = (string) $this->project->id;
    }
```

**Reply by joshhanley** [reply]
@d-shannon no worries! 🙂
